controllers/signupController.js

```diff
@@ -29,17 +25,26 @@ module.exports = {
         } catch(err){
             res.send({ err });
         }
+         
+        try { 
+            await bcrypt.hash(password, null, null, (err, hash) => {
+                user.password = hash;
+            });
 
-        // Save the new user
-        try {
+            const user = Object.assign(new User(), {
+                email,
+                username,
+            });
+            
```

Comment:
use object spread instead of Object.assign...its a bit cleaner
```
const user = {
   ...new User(),
   email,
   username,
};
```

Reply:
the spread operator is not working for me

returns
{
"err": {},
"success": false
}

Reply:
pull the most recent master, I added the fix for the spread operator

Reply:
what is the fix?